vscode-dotnet-runtime-library/src/Acquisition/GlobalInstallerResolver.ts

```diff
@@ -0,0 +1,376 @@
+    {
+        if(this.versionResolver.isNonSpecificMajorOrMajorMinorVersion(version))
+        {
+            Debugging.log(`The VersionResolver resolved the version to be major, or major.minor.`, this.eventStream);
+            const numberOfPeriods = version.split('.').length - 1;
+            const indexUrl = this.getIndexUrl(numberOfPeriods === 0 ? `${version}.0` : version);
+            const indexJsonData = await this.fetchJsonObjectFromUrl(indexUrl);
+            const fullySpecifiedVersionRequested = indexJsonData[this.releasesLatestSdkKey];
+            return [await this.findCorrectInstallerUrl(fullySpecifiedVersionRequested, indexUrl), fullySpecifiedVersionRequested];
+        }
+        else if(this.versionResolver.isNonSpecificFeatureBandedVersion(version))
+        {
+            Debugging.log(`The VersionResolver resolved the version to be a N.Y.XXX version.`, this.eventStream);
+            const fullySpecifiedVersion = await this.getNewestSpecificVersionFromFeatureBand(version);
+            return [
+                await this.findCorrectInstallerUrl(fullySpecifiedVersion, this.getIndexUrl(this.versionResolver.getMajorMinor(fullySpecifiedVersion))),
+                fullySpecifiedVersion
+            ];
+        }
+        else if(this.versionResolver.isFullySpecifiedVersion(version))
+        {
+            Debugging.log(`The VersionResolver resolved the version to be a fully specified version.`, this.eventStream);
+            const fullySpecifiedVersionRequested = version;
+            const indexUrl = this.getIndexUrl(this.versionResolver.getMajorMinor(fullySpecifiedVersionRequested));
+            return [await this.findCorrectInstallerUrl(fullySpecifiedVersionRequested, indexUrl), fullySpecifiedVersionRequested];
+        }
+
+        Debugging.log(`The VersionResolver could not resolve the version, version: ${version}.`, this.eventStream);
+        const err = new DotnetVersionResolutionError(new Error(`${this.badResolvedVersionErrorString} ${version}`), version);
+        this.eventStream.post(err);
+        throw err.error;
+    }
+
+    /**
+     *
+     * @remarks this function handles finding the right os, arch url for the installer.
+     * @param specificVersion the full, specific version, e.g. 7.0.301 to get.
+     * @param indexUrl The url of the index server that hosts installer downlod links.
+     * @returns The installer url to download.
+     */
+    private async findCorrectInstallerUrl(specificVersion : string, indexUrl : string) : Promise<string>
+    {
+        if(specificVersion === null || specificVersion === undefined || specificVersion === '')
+        {
+            const versionErr = new DotnetVersionResolutionError(new Error(`${this.badResolvedVersionErrorString} ${specificVersion}.`), specificVersion);
+            this.eventStream.post(versionErr);
+            throw versionErr.error;
+        }
+
+        const operatingSys : string = os.platform();
```

Comment:
nit: I might factor out all the logic to get the converted OS/ arch strings into helper methods, since this method is so long and that logic is theoretically somewhat separate.

Reply:
That's a good suggestion, I moved it all to FileUtilities :)